spec/models/user_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { User.new(name: 'name', photo: 'photo', bio: 'bio', posts_counter: 0) }
+  context 'initial object' do
+    it 'subject should be a valid user' do
+      expect(subject).to(be_valid)
+    end
+  end
+  context 'name attribute' do
+    it 'should not be blank' do
+      subject.name = ''
+      expect(subject).to_not(be_valid)
+    end
+  end
+  context 'posts_counter attribute' do
+    it 'should be an integer' do
+      subject.posts_counter = 'i'
+      expect(subject).to_not(be_valid)
+    end
+    it 'should be greate than or equal zero' do
+      subject.posts_counter = '-1'
+      expect(subject).to_not(be_valid)
+    end
+  end
+  context 'recent_posts method' do
+    it 'should return last three posts' do
+      expect(subject.recent_posts).to(eq(subject.posts))
+    end
+  end
+end
```

Comment:
Good job on your test coverage! Your attention to the required validations in the User model is commendable. You've written tests for the name and posts_counter validations, as well as the recent_posts method, which are crucial for the integrity of the User model.

### Room for Improvement:
However, I noticed that while we have tests for negative values and zero for `posts_counter`, we don't have explicit tests for positive values. It's essential to test the boundary conditions of our validations to make sure they behave correctly across the expected range of inputs.

```.txt
PostsCounter must be an integer greater than or equal to zero.
```
### Suggestion for Improvement:
Consider adding test cases for positive values to ensure comprehensive coverage. For instance, a test for `posts_counter` could be as follows:

```ruby
it 'is valid with a positive likes_counter value' do
    subject.posts_counter = '1'
    expect(post).to be_valid
end
```

Adding such tests will help in confirming that our model not only rejects invalid data but also accepts valid data as it should. This is just as important in TDD (Test-Driven Development) to prevent any false positives and to ensure the robustness of your application.

### Final Note:
Again, great work on your tests so far. By expanding the test cases to cover the full range of valid and invalid inputs, you will further solidify your application's reliability and maintainability.

Keep it up!

---